realm/realm-library/src/main/java/io/realm/SetValueOperator.java

```diff
@@ -340,17 +400,12 @@ boolean containsInternal(@Nullable Object o) {
     }
 
     @Override
-    boolean remove(@Nullable Object o) {
+    boolean removeInternal(@Nullable Object o) {
```

Comment:
Check type.

Reply:
It's done in the caller, I will add a comment.